### DISCUSSION TITLE
Ability to configure the target pull request branch

### DISCUSSION THREAD

**Discussion by mitch1995** [renovatebot/renovate#20391] [Suggest an Idea]
### What would you like Renovate to be able to do?

I have a use case where I would like to be able to configure the target pull request branch. This would differ from the `baseBranch`. For example, my current workflow is:

1. Branch from `stable`.
2. Open a pull request targeting `master`.

Currently I am not able to achieve this, as setting the `baseBranch` as `stable` causes the pull request to be opened targeting `stable` as well. 

### If you have any ideas on how this should be implemented, please tell us here.

Introduce a new property for `targetPrBranch`. The property can then be used when setting the target branch, e.g.
`targetBranch: config.targetPrBranch ?? config.baseBranch ?? ''`
https://github.com/renovatebot/renovate/blob/5caa59aaa151462c4c62a0d46633102f475078db/lib/workers/repository/update/pr/index.ts#L365

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
At first glance I don't like the sound of this. Why would you want it?

**Comment by mitch1995** [comment]
To support my current workflow; we have a `stable` branch that is guaranteed to be green, whereas `master` can at times be red. So whenever we checkout a new branch we will always branch off of `stable` and then merge into `master`, and then once `master` has been built and confirmed as working the changes are rolled from `master` -> `stable`

**Comment by rarkins** [comment]
Converted to an ideas discussion as I don't like this as a feature.. yet

**Comment by rarkins** [comment]
Supporting this capability could introduce difficult edge cases, including:
- The target branch may not have a shared history with the base branch (hopefully unlikely)
- The PR will be immediately conflicted with the target branch. Renovate has no choice but to never deconflict such PRs or continuously attempt to deconflict them (likely to happen)

**Comment by mitch1995** [comment]
> The target branch may not have a shared history with the base branch (hopefully unlikely)

Yes this is possible, but in this case the developer should not configure the `prTargetBranch` (so it would then just default to `baseBranch`). We could add a disclaimer in the docs to mention that the base and target branch should have a shared history.

> The PR will be immediately conflicted with the target branch. Renovate has no choice but to never deconflict such PRs or continuously attempt to deconflict them (likely to happen)

Not sure I understand this, why would it be immediately conflicted with the target branch if it shares the history?

<img width="255" alt="image" src="https://user-images.githubusercontent.com/33076858/218888945-f0522d0d-f71e-4003-a9bf-66053940a292.png">

The PR would be opened targeting master and should only include the one renovate dependency upgrade commit.

I've already validated this exact workflow via my current workaround: introduce a `postUpgradeTask` that changes the renovate PR target branch to `master` and prefixes the prTitle with `rebase!`. I haven't noticed any conflicted pull requests with this approach but it's quite a hacky workaround and requires constantly updating the prTitle to include `rebase!` after each rebase (since the `targetBranch` is no longer equal to the `baseBranch` so the branch is considered modified). I would love to avoid this and introduce support for configuring the target branch instead

**Reply by rarkins** [reply]
It's common that Renovate PRs touch the same or adjacent lines of code. The more that are merged, the higher the chance that other PRs have git conflicts

**Reply by mitch1995** [reply]
Isn't this already the case though? if multiple renovate PRs are modifying the same line, once the first one is merged the others would become conflicted. However this would be fixed in the next renovate execution when it rebases anyway

**Reply by rarkins** [reply]
The PR would be conflicted against the _target_ branch, not the _base_ branch. So even if it was rebased, it wouldn't help because it's rebased on top of the base branch.

**Reply by mitch1995** [reply]
Ahh I understand & you're right, that would happen. In my current use case I do not have multiple renovate PRs targeting the same package so have yet to experienced this
